Apps/WebClient/src/ClientApp/src/components/calendar/header.vue

```diff
@@ -141,7 +136,11 @@ export default class CalendarComponent extends Vue {
                 d.getFullYear() == date.getFullYear() &&
                 d.getMonth() == date.getMonth()
         );
-        this.dispatchEvent();
+    }
+
+    private dispatchEvent() {
+        let startDate = DateUtil.getMonthFirstDate(this.headerDate);
+        this.$emit("update:currentDate", startDate);
```

Comment:
How about updating startDate to currentDate?

Reply:
There is already a variable called currentDate.

Reply:
so I'm not sure why do we call this event with update:currentDate then? will it cause confusions with the existing currentDate? Regardless, I'm seeing that this line #143 wants to notify something about "update:currentDate" with a value of startDate, not sure if it logically makes sense though.

Reply:
update:currentDate syncs the variable passed in the properties currentDate. That is why the emmited event is named so.
https://vuejs.org/v2/guide/components-custom-events.html#sync-Modifier

Reply:
can we refactor the name of "currentDate" to something like "contextDate" or "selectedDate" as in most systems, current date means today's date.

Reply:
refactor done